### DISCUSSION TITLE
Cannot load my colorscheme plugin from local dev-folder

### DISCUSSION THREAD

**Discussion by goerz** [neovim/neovim#29412] [Q&A] [answered]
(Reposting from [Discourse](https://neovim.discourse.group/t/cannot-load-my-colorscheme-plugin-from-local-dev-folder/4963), since I realized that Discourse is "deprecated" in favor of GitHub Discussions):

I'm trying to point Lazy.nvim to a local checkout of my [personal colorscheme](https://github.com/goerz/goerz-colors.nvim) (`goerz`) so that I can develop it.

I have the [Lazy setup in my `init.lua` file](https://github.com/goerz/neovimrc/blob/138cde0e277583c12dced88d53618d303f37cf71/init.lua#L270). Also, [installing the colorscheme from Github](https://github.com/goerz/neovimrc/blob/master/lua/custom/plugins/goerz-colors.lua) works fine with the following configuration:

```lua
  { -- goerz color scheme
    'goerz/goerz-colors.nvim',
    lazy = false,
    priority = 1000, -- make sure to load this before all the other start plugins
  }
```

Now I've modified this with a `dir` and `dev` options:

```lua
 { -- goerz color scheme
    'goerz/goerz-colors.nvim',
    dir = "/home/goerz/Documents/Programming/goerz-colors.nvim/",
    dev = true,
    lazy = false,
    priority = 1000, -- make sure to load this before all the other start plugins
  }
```

I've made sure that the directory in `dir` exists and is accessible. Still, when I start `nvim`, I get an error like

```plain
Error detected while processing /home/goerz/.config/nvim/init.lua:
E5113: Error while calling lua chunk: /home/goerz/.config/nvim/init.lua:304: Vim:E185: Cannot find color scheme 'goerz'
stack traceback:
        [C]: in function 'colorscheme'
        /home/goerz/.config/nvim/init.lua:304: in main chunk
```

I've also tried commenting out the reference to GitHub (` 'goerz/goerz-colors.nvim',`).
How exactly do I tell Lazy.nvim to load a plugin from a local path? It seemed to me like I was following the instructions in the Lazy documentation. Is there something specific to colorschemes that would mess this up?

**Comment by goerz** [comment, marked as answer]
It seems to work if I leave out

```
dev = true,
```

**Reply by clason** [reply]
Sounds like this is a Lazy.nvim issue, so you should ask it over there.

**Reply by goerz** [reply]
Yup, I should have… Just like I only realized belatedly that Discourse isn't the best place to ask questions, I also didn't realize until half an hour ago that Lazy.nvim has its own Discussions on GitHub ;-)

Sorry, I'm pretty new to the Neovim online community. Although I've been using vim for 20 years, I'm only now getting around to really committing to Lua and all the modern niceties it brings.

**Reply by goerz** [reply]
For what it's worth, I thought that `dev = true` in combination with `dir` would be an "editable install" (similar to how, e.g., in Python world, there's `pip install .` and `pip install -e .`). But instead, it seems giving `dir` is always "editable" (modifying the files in the `dir` and then doing `Lazy reload <plugin>` sees the updated code). It seems what `dev = true` is actually for is to clone the plugin from its GitHub address to `config.dev.path`. So my understanding is that `dev = true` is basically the same as `dir` but with an automatic location. That makes the `dev` and `dir` options mutually exclusive.
